### PR TITLE
Adds proxy support for API calls to BrowserStack

### PR DESCRIPTION
**Pull request by shirish87**
* Bumps `browserstack` module to latest version
* Passes proxy settings to the `browserstack` module for making API requests using the configured proxy

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -90,11 +90,19 @@ var createBrowserStackClient = function (/* config.browserStack */config) {
...
+  var options = {
     username: env.BROWSER_STACK_USERNAME || config.username,
     password: env.BROWSER_STACK_ACCESS_KEY || config.accessKey
-  })
+  }
+
+  if (config.proxyHost && config.proxyPort) {
+    var proxyAuth = (config.proxyUser && config.proxyPass)
+      ? (config.proxyUser + ':' + config.proxyPass + '@') : ''
+    options.proxy = 'http://' + proxyAuth + config.proxyHost + ':' + config.proxyPort
```
Comment (dignifiedquire) [inline index.js]: What about `https` based proxies?

Comment (shirish87) [inline index.js]: yes, i wasn't sure how to handle that. pass another `proxyProtocol` param?

Comment (dignifiedquire) [inline index.js]: Yes that would be good, but it should default to `http` when not set. 

Also can you please add some details about this to the readme so people know it's there :)

Comment (shirish87) [inline index.js]: Done. I've made some changes to the README too. Please have a look.
Thanks!

**Comment by shirish87** [comment]
@dignifiedquire, have you found a chance to review this? : )

**Comment by dignifiedquire** [comment]
Thanks, sorry for the delay :octocat:

**Comment by shirish87** [comment]
any chance of a release soon? :)

**Pull request merged**